Intersect.Client/Interface/Game/DescriptionWindows/DescriptionWindowBase.cs

```diff
@@ -0,0 +1,160 @@
+            if (loadLayout)
+            {
+                component.LoadLayout();
+            }
+            mComponents.Add(component);
+            return component;
+        }
+
+        /// <summary>
+        /// Adds a <see cref="DescriptionComponent"/> to the current window.
+        /// </summary>
+        /// <param name="name">The name of the component.</param>
+        /// <param name="loadLayout">Should we load the layour json file automatically?</param>
+        /// <returns>Returns a new instance of the <see cref="DescriptionComponent"/> class</returns>
+        public DescriptionComponent AddDescription(string name = "DescriptionWindowDescription", bool loadLayout = true)
+        {
+            var component = new DescriptionComponent(mContainer, name);
+            if (loadLayout)
+            {
+                component.LoadLayout();
+            }
+            mComponents.Add(component);
+            return component;
+        }
+
+        /// <summary>
+        /// Adds a <see cref="RowContainerComponent"/> to the current window.
+        /// </summary>
+        /// <param name="name">The name of the component.</param>
+        /// <returns>Returns a new instance of the <see cref="DescriptionComponent"/> class</returns>
+        public RowContainerComponent AddRowContainer(string name = "DescriptionWindowRowContainer")
+        {
+            var component = new RowContainerComponent(mContainer, name);
+            mComponents.Add(component);
+            return component;
+        }
+
+        /// <summary>
+        /// Positions a component correctly on the current window.
+        /// </summary>
+        /// <param name="component">The <see cref="ComponentBase"/> to place.</param>
+        private void PositionComponent(ComponentBase component)
+        {
+            if (component == null)
+            {
+                return;
+            }
+
+            component.SetPosition(component.X, mComponentY);
+            mComponentY += component.Height;
+        }
+
+        /// <summary>
+        /// Position and resize all components properly for display.
+        /// </summary>
+        protected void FinalizeWindow()
+        {
+            // Reset our componentY so we start from scratch!
+            mComponentY = 0;
+
+            // Correctly set our container width to the largest child start with, this way our other child components will have a width to work with.
+            mContainer.SizeToChildren(true, false);
+
+            // Resize and relocate our components to properly display on our window.
+            foreach (var component in mComponents)
+            {
+                component.CorrectWidth();
+                PositionComponent(component);
+            }
+
+            // Correctly set our container height so we display everything.
+            mContainer.SizeToChildren(false, true);
+        }
+
+        /// <inheritdoc/>
+        public override void SetPosition(int x, int y)
```

Comment:
[Click here to see the comment/suggested change.](https://github.com/AscensionGameDev/Intersect-Engine/pull/970#issuecomment-1045175828)

Reply:
![imagen](https://user-images.githubusercontent.com/17498701/157314102-a7d6d9e2-5f65-4fa3-9cf2-9aac5bcb8d55.png)

This could be handled by enums perhaps?

Reply:
![imagen](https://user-images.githubusercontent.com/17498701/157314102-a7d6d9e2-5f65-4fa3-9cf2-9aac5bcb8d55.png)

This could be handled by enums perhaps?
It would be cool to have an enum that allows the description windows to follow the cursor position as well 💡